### ISSUE TITLE
[Readme Markdown Parser] Lots of warnings today

### ISSUE THREAD

**Issue by Phasar40** [netmix/radio-station#285]
Deprecated: Array and string offset access syntax with curly braces is deprecated in D:\xampp\apps\wordpress\htdocs\wp-content\plugins\radio-station\reader.php on line 1151

Deprecated: Array and string offset access syntax with curly braces is deprecated in D:\xampp\apps\wordpress\htdocs\wp-content\plugins\radio-station\reader.php on line 1491

Deprecated: Array and string offset access syntax with curly braces is deprecated in D:\xampp\apps\wordpress\htdocs\wp-content\plugins\radio-station\reader.php on line 1493

Deprecated: Methods with the same name as their class will not be constructors in a future version of PHP; Markdown_Parser has a deprecated constructor in D:\xampp\apps\wordpress\htdocs\wp-content\plugins\radio-station\reader.php on line 104

Deprecated: Array and string offset access syntax with curly braces is deprecated in D:\xampp\apps\wordpress\htdocs\wp-content\plugins\radio-station\reader.php on line 1714

Deprecated: Array and string offset access syntax with curly braces is deprecated in D:\xampp\apps\wordpress\htdocs\wp-content\plugins\radio-station\reader.php on line 1716

Deprecated: Array and string offset access syntax with curly braces is deprecated in D:\xampp\apps\wordpress\htdocs\wp-content\plugins\radio-station\reader.php on line 1979

Deprecated: Array and string offset access syntax with curly braces is deprecated in D:\xampp\apps\wordpress\htdocs\wp-content\plugins\radio-station\reader.php on line 1979

Deprecated: Array and string offset access syntax with curly braces is deprecated in D:\xampp\apps\wordpress\htdocs\wp-content\plugins\radio-station\reader.php on line 1988

Deprecated: Array and string offset access syntax with curly braces is deprecated in D:\xampp\apps\wordpress\htdocs\wp-content\plugins\radio-station\reader.php on line 2026

Deprecated: Array and string offset access syntax with curly braces is deprecated in D:\xampp\apps\wordpress\htdocs\wp-content\plugins\radio-station\reader.php on line 2026

Deprecated: Array and string offset access syntax with curly braces is deprecated in D:\xampp\apps\wordpress\htdocs\wp-content\plugins\radio-station\reader.php on line 2045

Deprecated: Array and string offset access syntax with curly braces is deprecated in D:\xampp\apps\wordpress\htdocs\wp-content\plugins\radio-station\reader.php on line 2046

Deprecated: Array and string offset access syntax with curly braces is deprecated in D:\xampp\apps\wordpress\htdocs\wp-content\plugins\radio-station\reader.php on line 2150

Deprecated: Array and string offset access syntax with curly braces is deprecated in D:\xampp\apps\wordpress\htdocs\wp-content\plugins\radio-station\reader.php on line 2162

Deprecated: Array and string offset access syntax with curly braces is deprecated in D:\xampp\apps\wordpress\htdocs\wp-content\plugins\radio-station\reader.php on line 2162

Deprecated: Array and string offset access syntax with curly braces is deprecated in D:\xampp\apps\wordpress\htdocs\wp-content\plugins\radio-station\reader.php on line 2173

Deprecated: Array and string offset access syntax with curly braces is deprecated in D:\xampp\apps\wordpress\htdocs\wp-content\plugins\radio-station\reader.php on line 2174

Deprecated: Array and string offset access syntax with curly braces is deprecated in D:\xampp\apps\wordpress\htdocs\wp-content\plugins\radio-station\reader.php on line 2532

Deprecated: Array and string offset access syntax with curly braces is deprecated in D:\xampp\apps\wordpress\htdocs\wp-content\plugins\radio-station\reader.php on line 2850

Deprecated: Methods with the same name as their class will not be constructors in a future version of PHP; MarkdownExtra_Parser has a deprecated constructor in D:\xampp\apps\wordpress\htdocs\wp-content\plugins\radio-station\reader.php on line 1584

**Comment by tonyzeoli** [comment]
@majick777 Please take a look at this, thanks.

**Comment by tonyzeoli** [comment]
@Phasar40 Is this with error reporting turned on?

**Comment by majick777** [comment]
@Phasar40 Thanks for reporting. These are all from WordPress' Markdown Parser class... which it seems like it hasn't been updated in a long time (2013?!) - as I can't seem to find a more modern version. 

In any case, I modernized it by replacing the curly braces with square ones and changing the constructor to the `__construct` method. So those warnings should no longer be appearing admin side or in your debug log. Added to develop branch for 2.3.3.7 release.
